SIPS/sip-25.md

```diff
@@ -0,0 +1,225 @@
+
+
+## Motivation
+
+The integration of MetaMask's MultiChain API and protocol Snaps necessitates a standardized approach for routing RPC requests across various blockchain networks. By establishing a standard interface for RPC routing Snaps, we aim to abstract the complexity of non-EVM protocol support away from the MetaMask engineering team, ensuring a seamless and scalable interaction model for decentralized applications (dApps). Because MetaMask is not equipped to interpret requests from non-EVM networks and, for instance, identify which of these requests require signatures or, for methods that do require signatures, which parameters correspond to the account that should sign a transaction, the RPC routing Snap will provide logic for interpreting and routing these requests accordingly.
+
+This approach allows for greater flexibility and extensibility, enabling the MetaMask ecosystem to support a wide range of blockchain networks without requiring changes to the core wallet functionality.
+
+This SIP outlines the key responsibilities and minimum interface requirements for RPC routing Snaps, enabling developers to build and deploy routing Snaps for specific blockchain networks.
+
+## Specification
+
+### Overview
+
+RPC routing Snaps will serve as intermediaries for specific [CAIP-2](https://github.com/ChainAgnostic/CAIPs/blob/main/CAIPs/caip-2.md) identifiers, registering themselves with MetaMask and managing account Snap registrations for their respective networks. They will interpret CAIP-27 requests targeting non-EVM networks made to their registered CAIP-2 identifier, route these requests, facilitate user-specified account Snap options for signing requests, and broadcast signed transactions to the network.
+
+### Key Responsibilities
+
+1. **Registration with MetaMask**:
+   - RPC routing Snaps must register with MetaMask for the CAIP-2 identifiers they can serve.
+
+2. **Account Snap Registration**:
+   - RPC routing Snaps will expose an interface for account Snaps which serve the same CAIP-2 identifier to register themselves with the routing Snap.
+   - Account Snaps must declare the signing methods they support with the routing Snap.
+
+3. **Handling CAIP-27 Requests**:
+   - RPC routing Snaps will interpret incoming CAIP-27 requests, identify the network-specific RPC request format, and route to the appropriate account Snap.
+   - If the request does not specify an account the routing Snap must provide UI for users to select from among the eligible signer Snaps for signing the request.
```

Comment:
Is there a specific reason we would assume the Snap being the best facilitator of this over MetaMask itself?